### DISCUSSION TITLE
Magic property to select the current element

### DISCUSSION THREAD

**Discussion by leecolarelli** [alpinejs/alpine#945] [3. Feature ideas]
Hi all,

I recently built an accordion with alpine and tailwind, which sets a max height of 0 on the content, and then on click sets a max height to match the content with `scrollHeight`.

If you look at both of the pens linked below, the first uses $el to set the content max height to match the root element. This works because the content is using template and x-for, but if you remove the template/x-for and just set hard coded content, then the $el.scrollHeight is not detected correctly. Instead, it detects the scrollHeight as the height of the root element with content max-h-0.

So in my version (second pen) i needed to set the max height by using x-ref and $refs to both label and select the current element with
```
x-ref="foo" x-bind:style="search_selected === true ? `max-height:  ${ $refs.foo.scrollHeight }px` : ``"
```

Therefore, wouldn't it be cleaner here to just have a magic property to select the current element?

https://codepen.io/hmaesta/pen/MWKoGEm
https://codepen.io/leecolarelli/pen/PoGjWZy

**Comment by KevinBatdorf** [comment]
> Therefore, wouldn't it be cleaner here to just have a magic property to select the current element?

Isn't that what `x-ref`/`$refs` solves?

**Reply by leecolarelli** [reply]
Not in this instance, as it has to be declared and then used, whereas $el doesn’t have to be declared first

**Reply by HugoDF** [reply]
@leecolarelli @KevinBatdorf is this even possible? Last time I checked magic properties get initialised per component (the root element is passed to the magic property initialiser) so don't think it'll be possible to do

I think Alpine v3 might have this behaviour, `$el` will be the element on which the directive is being evaluated

**Reply by KevinBatdorf** [reply]
_Might_ be possible in a similar way to how `$event` is added, but it's pretty easily solved using `$refs`.  It's also not something that seems to be in demand. It would also likely need to be implemented in multiple directives. 

I think `$this` would make sense as the helper name though if someone wants to PR it :)

**Reply by Elliotclyde** [reply]
I had a situation where referring to the children elememts without $ref could've saved quite a lot code. I would be all for a $this helper.

**Reply by ryangjchandler** [reply]
I agree that a `$this` helper would be beneficial as a shortcut since there's no need to keep refs up to date. 

@calebporzio Is this something we can consider for v3?

**Reply by HugoDF** [reply]
> I agree that a `$this` helper would be beneficial as a shortcut since there's no need to keep refs up to date.
> 
> @calebporzio Is this something we can consider for v3?

I think Caleb is making changes to `$el` for v3
